src/runtime/micro/micro_session.h

```diff
@@ -174,29 +204,24 @@ class MicroSession : public ModuleNode {
   /*! \brief array of memory allocators for each on-device section */
   std::shared_ptr<MicroSectionAllocator>
       section_allocators_[static_cast<size_t>(SectionKind::kNumKinds)];
-  /*! \brief total number of bytes of usable device memory for this session */
-  size_t memory_size_;
-  /*! \brief uTVM runtime binary info */
-  BinaryInfo runtime_bin_info_;
-  /*! \brief path to uTVM runtime source code */
-  std::string runtime_binary_path_;
-  /*! \brief offset of the runtime entry function */
-  DevBaseOffset utvm_main_symbol_;
-  /*! \brief offset of the runtime exit breakpoint */
-  DevBaseOffset utvm_done_symbol_;
+  /*! \brief number of bytes in a word on the target device */
+  size_t word_size_;
+  /*! \brief whether the target device requires a thumb-mode bit on function addresses
+   *
+   * ARM and other manufacturers use the lowest bit of a function address to determine
+   * whether it's a "thumb mode" function.  The Thumb ISA is more restricted, but
+   * results in more compact binaries.
+   */
+  bool thumb_mode_;
```

Comment:
For Cortex-M based devices, the only variant of the AArch32 ISA supported is Thumb (i.e. T16 / T32) so that's always default. Thus I don't think you need to worry about this. It's only on older cores (i.e. CPUs implementing Armv6 (except Armv6T2), Armv4t, Armv5t*,  or with cores from other profiles that you need to worry between A32 and T16 / T32.)

Reply:
see my earlier comment about the `thumb_mode` flag